src/Adapter/Product/Validate/ProductValidator.php

```diff
@@ -198,7 +198,7 @@ private function validateStock(Product $product): void
         } else {
             $this->validateClassicStock($product);
         }
-
+        //@todo: quantity validation missing. product->quantity is not defined in Product.php $definition so its always 0 & cannot be validated by legacy
```

Comment:
Then we should add a `StockAvailableValidator` to validate it on the `StockAvailable` this data is quite sensitive we shouldn't leave it without any checks

Reply:
yep as discussed in slack.
It could be done in both Product and Combination in another PR.

Reply:
Maybe this todo comment is useless now?